src/chart/bar/BarChart.tsx

```diff
@@ -203,80 +236,174 @@ const NeoBarChart = (props: ChartProps) => {
   };
 
   // Fixing canvas bug, from https://github.com/plouc/nivo/issues/2162
+  // SVGGraphicsElement.getBBox
   HTMLCanvasElement.prototype.getBBox = function tooltipMapper() {
     return { width: this.offsetWidth, height: this.offsetHeight };
   };
 
+  // positionLabel == 'off' ? {} :
   const extraProperties = positionLabel == 'off' ? {} : { barComponent: BarComponent };
   const canvas = data.length > 30;
   const BarChartComponent = canvas ? ResponsiveBarCanvas : ResponsiveBar;
+
+  // For adaptable item length in the legend
+  const maxKeyLength = Math.max(...keys.map((key) => key.length));
+  const baseItemWidth = 40; // Some base width for color box and padding
+  const charWidthEstimate = 5; // An estimate of how wide each character is, you might need to adjust this based on font size and type
+  const itemWidthConst = baseItemWidth + maxKeyLength * charWidthEstimate;
+  const adaptableWidth =
+    marginLeft +
+    marginRight +
+    data.length * barWidth * 4 +
+    (data.length - 1) * 4 +
+    (data.length - 1) * innerPadding * 4;
+
+  // Container to make the chart scroll horizontally
+  const scrollableWrapperStyle: React.CSSProperties = {
```

Comment:
add a comment here explaining why / what we do to calculate the dimensions.
Also where the numbers `200`, `5`, `18`, `1.2`,  come from in the calculations below, it's not clear what they represent.